utility/src/main/java/org/oppia/android/util/namevalidator/NameValidator.kt

```diff
@@ -0,0 +1,39 @@
+package org.oppia.android.util.namevalidator
+
+class NameValidator {
+  companion object {
+    private const val MATCH_INTERNATIONAL_LETTERS_REJECT_SPACES =
+      "^.[\\w\\u00BF-\\u1FFF\\u2C00-\\uD7FF.'\\-]+\$"
+
+    private const val REJECT_SYMBOLS = "^[^#*!@\$%^&()_+=\\\\|\\]\\[\":;?/><,`~{}]*\$"
+
+    private const val REJECT_REPEATED_USE_OF_ALLOWED_SYMBOLS = "[\\.'-]{2}"
+
+    fun validate(name: String): Boolean {
+      return (
+        notEmptyNoSpacesAndContainsLetters(name) &&
+          noNumbers(name) &&
+          noSymbols(name) &&
+          noRepeatedUseOfAllowedSymbols(name)
+        )
+    }
+
+    private fun notEmptyNoSpacesAndContainsLetters(name: String): Boolean {
```

Comment:
Method names should be verb phrases, e.g.: ``containsLettersAndNoSpaces``.

Ditto for the others.

You could also consider making these string extension function, e.g.:

```kotlin
private fun String.containsLettersAndNoSpaces(): Boolean =
  matches(Regex(MATCH_INTERNATIONAL_LETTERS_REJECT_SPACES))
```